### PR TITLE
Use logtest.AssertRecordEqual in logrus bridge

### LINKED ISSUE #5851: Use logtest.AssertRecordEqual in otellogrus test
_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5847#discussion_r1663893751_

### PR DESCRIPTION
**Pull request by dmathieu**
Closes #5851

### CODE REVIEW

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by pellared** [comment]
@open-telemetry/go-approvers PTAL (needs an approval from a non-Splunk employee)

**Review by dashpole — APPROVED** [review]

**Pull request merged**
